subprojects/enterprise/src/main/java/org/gradle/internal/tasks/TaskProperties.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.tasks;
+
+import java.util.Set;
+
+public interface TaskProperties {
+
+    Set<InputFileProperty> getInputFiles();
+
+    Set<OutputFileProperty> getOutputFiles();
```

Comment:
We should probably capture non-file inputs as well, no?

Reply:
We could add them but we (currently) don't need them.

Reply:
I think it really depends on our understanding of this interface in the `enterprise` project. I can see two cases:

* the interface is there for TD to use and it offers what TD requires (or more to be safe in the future)
* the interface is as generic as possible and TD is just one party (albeit the only one, currently) using it

Personally, I would go for version 1 here.